cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -207,6 +208,183 @@ func Run(c *config.CompletedConfig, stopCh <-chan struct{}) error {
 		}
 	}
 
+	// If leader migration is enabled, use the redirected initialization
+	// otherwise, make absolutely NO change to the initialization process
+	if leadermigration.FeatureEnabled() && c.ComponentConfig.Generic.LeaderMigrationEnabled &&
+		c.ComponentConfig.Generic.LeaderElection.LeaderElect {
+
+		klog.Infof("starting leader migration")
```

Comment:
Please extract this into its own function. We should not be adding two hundred lines of code for an alpha feature directly into the main KCM run() method.

Reply:
Done.